tests/data/test_temporal_random_walker.py

```diff
@@ -17,21 +17,32 @@
 import pandas as pd
 import pytest
 import numpy as np
-import networkx as nx
 from stellargraph.data.explorer import TemporalRandomWalk
 from stellargraph.core.graph import StellarGraph
 
 
-@pytest.fixture()
-def temporal_graph():
-    nodes = [1, 2, 3, 4, 5, 6]
-    edges = [(1, 2, 5), (2, 3, 2), (2, 4, 10), (4, 5, 3), (4, 6, 12)]
+def create_temporal_graph(nodes, edges):
     edge_cols = ["source", "target", "weight"]
     return StellarGraph(
         nodes=pd.DataFrame(index=nodes), edges=pd.DataFrame(edges, columns=edge_cols),
     )
 
 
+@pytest.fixture()
+def temporal_graph():
+    nodes = [1, 2, 3, 4, 5, 6]
+    edges = [(1, 2, 5), (2, 3, 2), (2, 4, 10), (4, 5, 3), (4, 6, 12)]
+    return create_temporal_graph(nodes, edges)
+
+
+def temporal_graph_negative_times(num_edges):
+    nodes = [1, 2, 3, 4, 5, 6]
+    edges = [
```

Comment:
could this be something like:

```suggestion
    edges = np.vstack([
        np.random.choice(nodes, size=num_edges), 
        np.random.choice(nodes, size=num_edges), 
        -np.ones(num_edges),
    ]) 
```

Reply:
Thanks! Actually I think this should also work
```python
edges = np.hstack([np.random.choice(nodes, size=(num_edges, 2)), -np.ones((5,1))])
```

Reply:
nice!